examples/nirfmxnr/acp-single-carrier.py

```diff
@@ -0,0 +1,325 @@
+#
+# Getting Started:
+#
+# To run this example, install "RFmx NR" on the server machine.
+# Link: https://www.ni.com/en-us/support/downloads/software-products/download.rfmx-nr.html
+#
+# For instructions on how to use protoc to generate gRPC client interfaces, see our "Creating a gRPC Client" wiki page.
+# Link: https://github.com/ni/grpc-device/wiki/Creating-a-gRPC-Client
+#
+# Refer to the NI-RFmxNR gRPC Wiki for the latest C Function Reference:
+# Link: https://github.com/ni/grpc-device/wiki/NI-RFmxNR-C-Function-Reference
+#
+# Running from command line:
+#
+# Server machine's IP address, port number, and physical channel name can be passed as separate command line arguments.
+#   > python acp-single-carrier.py <server_address> <port_number> <physical_channel_name>
+# If they are not passed in as command line arguments, then by default the server address will be "localhost:31763", with "SimulatedDevice" as the resource name
+
+
+import sys
+
+import grpc
+import nirfmxnr_pb2 as nirfmxnr_types
+import nirfmxnr_pb2_grpc as grpc_nirfmxnr
+
+server_address = "localhost"
+server_port = "31763"
+session_name = "RFmxNRSession"
+
+# Resource name and options for a simulated 5663 client.
+resource = "SimulatedDevice"
+options = "Simulate=1,DriverSetup=Model:5663"
+
+# Read in cmd args
+if len(sys.argv) >= 2:
+    server_address = sys.argv[1]
+if len(sys.argv) >= 3:
+    server_port = sys.argv[2]
+if len(sys.argv) >= 4:
+    resource = sys.argv[3]
+    options = ""
+
+# Create a gRPC channel + client.
+channel = grpc.insecure_channel(f"{server_address}:{server_port}")
+client = grpc_nirfmxnr.NiRFmxNRStub(channel)
+instr = None
+
+
+# Raise an exception if an error was returned
+def raise_if_error(response):
+    if response.status != 0:
+        error_response = client.GetError(
+            nirfmxnr_types.GetErrorRequest(
+                instrument=instr,
+            )
+        )
+        raise Exception(f"Error: {error_response.error_description}")
+
+    return response
+
+
+try:
+    auto_level = True
+
+    initialize_response = raise_if_error(
+        client.Initialize(
+            nirfmxnr_types.InitializeRequest(
+                session_name=session_name, resource_name=resource, option_string=""
+            )
+        )
+    )
+    instr = initialize_response.instrument
+    raise_if_error(
+        client.CfgFrequencyReference(
+            nirfmxnr_types.CfgFrequencyReferenceRequest(
+                instrument=None,
+                channel_name="",
+                frequency_reference_source_mapped=nirfmxnr_types.FREQUENCY_REFERENCE_SOURCE_ONBOARD_CLOCK,
+                frequency_reference_frequency=10.0e6,
+            )
+        )
+    )
+    raise_if_error(
+        client.SetAttributeString(
+            nirfmxnr_types.SetAttributeStringRequest(
+                instrument=instr,
+                selector_string="",
+                attribute_id=nirfmxnr_types.NIRFMXNR_ATTRIBUTE_SELECTED_PORTS,
+                attr_val_raw="",
```

Comment:
I would image attr_val is also a string here?

Reply:
SetAttributeString allows either `attr_val_raw` or `attr_val_mapped`, wherein the latter is a mapped value like FREQUENCY_REFERENCE_SOURCE_ONBOARD_CLOCK (whose value is 1). There's no `attr_val` for string attributes.